src/Device/deviceLister.ts

```diff
@@ -367,10 +366,42 @@ export const startWatchingDevices =
         });
     };
 
-const getAutoSelectDeviceCLISerial = () => {
+const getAutoSelectDeviceCLIProperty = (
+    property: string,
+    findDevice: (value: string) => Device | undefined
+) => {
     const { argv } = process;
-    const serialIndex = argv.findIndex(arg => arg === '--deviceSerial');
-    return serialIndex > -1 ? argv[serialIndex + 1] : undefined;
+    const index = argv.findIndex(arg => arg === property);
+    return index > -1
+        ? { index, device: findDevice(argv[index + 1]) }
+        : undefined;
+};
+
+const getAutoSelectDevice = (devices: Device[]) => {
+    const serialNumber = getAutoSelectDeviceCLIProperty('--deviceSerial', sn =>
+        devices.find(device => device.serialNumber === sn)
+    );
+    const serialPort = getAutoSelectDeviceCLIProperty('--comPort', portPath =>
+        devices.find(device =>
+            device.serialPorts?.find(port => port.comName === portPath)
+        )
+    );
+
+    if (serialNumber && serialPort) {
+        if (serialNumber.index > serialPort.index) {
+            return serialPort.device;
+        }
+
+        return serialNumber.device;
+    }
```

Comment:
In the launcher you, as discussed, do not handle the case if both serial number and port are given (instead we print an error). Here you do handle the case, with the rule that the later command line argument takes precedence. 

Both would be fine for me: Either keep it here like it is or just ignore the case here (by removing those 7 lines above, which would effectively give the serial number precedence if the impossible case happens that both are given).

Reply:
So we have three cases here 

- user opens app from cli with -- --deviceSerial ... --comPort ... -> in this case we close app immidiatly

- user opens app from cli with -- --deviceSerial ... e.g. cellular monitor --> app will open and have the -- --deviceSerial  in its argv, if user then runs click the button to open serial terminal and that will need to pass device -- --comPort launcher is already open (with no window) and we need to make sure that we pass the `mergeAdditionalArguments` first to always use what ever the openApp needs to do dance the order above. 

- user opens app from cli with -- --deviceSerial ... --> app will open and have the -- --deviceSerial  in its argv, if user then runs -- --comPort launcher is already open (with no window) and we need to make sure that we pass the arguments from the second CLI instance first as that is what we need to use .... having said this I am unable to verify this last case ...